crates/router/src/core/mandate.rs

```diff
@@ -157,8 +157,19 @@ pub async fn update_connector_mandate_id(
     merchant_account: String,
     mandate_ids_opt: Option<api_models::payments::MandateIds>,
     resp: Result<types::PaymentsResponseData, types::ErrorResponse>,
+    payment_method_id: Option<String>,
 ) -> RouterResponse<mandates::MandateResponse> {
-    let connector_mandate_id = Option::foreign_try_from(resp)?;
+    let mandate_details = Option::foreign_from(resp);
+
+    let connector_mandate_id = mandate_details
+        .clone()
+        .map(|md| {
+            Encode::<types::MandateReference>::encode_to_value(&md)
+                .change_context(errors::ApiErrorResponse::MandateNotFound)
```

Comment:
this error should be different, raise `InternalServerError` with appropriate printable